<?xml version="1.0"?>

<launch>

    <arg name="camera" default="camera" />

    <!-- start sensor-->
    <include file="$(find openni2_launch)/launch/openni2.launch">
        <arg name="camera" default="$(arg camera)"/>
    </include>

    <!-- run pointcloud_to_laserscan node -->
    <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan" name="pointcloud_to_laserscan">

        <param name="use_inf" value="true"/>
        <param name="use_concurrency" value="true"/>

        <param name="target_frame" value="base_link"/>
        <remap from="cloud_in" to="$(arg camera)/depth_registered/points_processed"/>
        <remap from="scan" to="$(arg camera)/scan"/>

    </node>

</launch>